e2e/tests/008.sh

```diff
@@ -29,78 +29,123 @@ source "${LIBDIR}/k8s.sh"
 
 kubeconfig="$HOME/.kube/config"
 
-
 function _check_scale {
     local metric=$1
     local previous=$2
     local current=$3
-    
+
     echo "UPF - Comparing the new $metric after scaling"
-    if [ "$previous" -le  "$current" ]; then
+    if [ "$previous" -le "$current" ]; then
         echo "UPF $metric scaling Failed"
         exit 1
     fi
     echo "UPF - $metric Pod Scaling Successful"
 }
 
+function _get_first_container_cpu {
+    local kubeconfig=$1
+    local namespace=$2
+    local pod_id=$3
+
+    # we probably need to convert these to some uniform units
+    kubectl --kubeconfig $kubeconfig get pods $pod_id -n $namespace -o jsonpath='{range .spec.containers[*]}{.resources.requests.cpu}{"\n"}{end}' | head -1 | sed 's/m$//'
+}
+
+function _get_first_container_memory {
+    local kubeconfig=$1
+    local namespace=$2
+    local pod_id=$3
+
+    # we probably need to convert these to some uniform units
+    kubectl --kubeconfig $kubeconfig get pods $pod_id -n $namespace -o jsonpath='{range .spec.containers[*]}{.resources.requests.memory}{"\n"}{end}' | head -1 | sed 's/[GM]i$//'
+}
```

Comment:
What about removing some duplicity?

```suggestion
function _get_first_container_resource {
    local kubeconfig=$1
    local namespace=$2
    local pod_id=$3
    local type=$4

    # we probably need to convert these to some uniform units
    kubectl --kubeconfig $kubeconfig get pods $pod_id -n $namespace -o jsonpath="{.items[0].spec.containers[0].resources.requests.$type}"
}

function _get_first_container_cpu {
    _get_first_container_resource "$1" "$2" "$3" "cpu" | sed 's/m$//'
}

function _get_first_container_memory {
    _get_first_container_resource "$1" "$2" "$3" "memory" | sed 's/[GM]i$//'
}
```

Reply:
Sure. I think eventually we can have something like this in the k8s.sh library, we will need it for other tests.

In fact, rather than changing this PR, can you build functions in that library to get CPU and memory, and normalize them to some standard units for each? Because right now if the unit shifts from say Mi to Gi, the test will fail.

Once those library routines are ready we can use them in this script.

Reply:
The could also potential either sum all container values, or let the user specify a container name in addition to a pod name.

Reply:
Let me try to investigate something about it.